radicle-ssh/src/agent/client.rs

```diff
@@ -231,26 +229,23 @@ impl<S: ClientStream> AgentClient<S> {
     }
 
     /// Ask the agent to sign the supplied piece of data.
-    pub fn sign_request<K>(&mut self, public: &K, data: Buffer) -> Result<Signature, Error>
+    pub fn sign<K>(&mut self, public: &K, data: &[u8]) -> Result<Signature, Error>
     where
         K: Encodable + fmt::Debug,
     {
-        self.prepare_sign_request(public, &data);
-        self.stream.read_response(&mut self.buf)?;
+        let req = self.prepare_sign_request(public, data);
+        let resp = self.stream.request(&req)?;
 
-        if !self.buf.is_empty() && self.buf[0] == msg::SIGN_RESPONSE {
-            let mut signature: Signature = [0; 64];
-            self.write_signature(&mut signature)?;
-
-            Ok(signature)
-        } else if self.buf[0] == msg::FAILURE {
+        if !resp.is_empty() && resp[0] == msg::SIGN_RESPONSE {
+            self.read_signature(&resp)
```

Comment:
This looks wrong? self.write_signature() was changed to self.read_signature()

Reply:
Ah, no it's because the method was changed and renamed to return a signature instead of writing one to a buffer:

https://github.com/radicle-dev/heartwood/pull/74/files/f5212405c52a1390b3f0af63b4b482deeba84761#diff-29636161e3b27caf926633f4a658b6b740f79ba222e44e8f22e690e91e24a3e0R274